snuba/datasets/metrics_processor.py

```diff
@@ -37,14 +45,29 @@ def process_message(
             "org_id": message["org_id"],
             "project_id": message["project_id"],
             "metric_id": message["metric_id"],
-            "metric_type": "set",
             "timestamp": timestamp,
             "tags.key": keys,
             "tags.value": values,
-            "set_values": message["value"],
+            **self._process_values(message),
             "materialization_version": 0,
             "retention_days": message["retention_days"],
             "partition": metadata.partition,
             "offset": metadata.offset,
         }
         return InsertBatch([processed], None)
+
+
+class SetsMetricsProcessor(MetricsProcessor):
+    def _should_process(self, message: Mapping[str, Any]) -> bool:
+        return message["type"] is not None and message["type"] == "s"
+
+    def _process_values(self, message: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {"set_values": message["value"]}
```

Comment:
Is it worth doing some validation here? At least while the project is getting started?

Reply:
Ok, the clickhouse message would be quite cryptic (it would complain a field is missing or invalid without telling which storage is affected) So Will add an assertion.

Reply:
Done